zaza/openstack/charm_tests/magnum/setup.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+#
+# Copyright 2019 Canonical Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import tenacity
+import os
+
+import zaza.model
+import zaza.openstack.utilities.openstack as openstack_utils
+
+
+@tenacity.retry(stop=tenacity.stop_after_attempt(3),
+                wait=tenacity.wait_exponential(
+                multiplier=1, min=2, max=10))
+def domain_setup(application_name='magnum'):
+    """Run required action for a working Magnum unit."""
+    # Action is REQUIRED to run for a functioning magnum deployment
+    logging.info('Running domain-setup action on magnum unit...')
+    unit = zaza.model.get_units(application_name)[0]
+    zaza.model.block_until_unit_wl_status(unit.entity_id, "active")
+    zaza.model.run_action(unit.entity_id, "domain-setup")
+    zaza.model.block_until_unit_wl_status(unit.entity_id, "active")
+
+
+def add_image(image_url=None):
+    """Upload the operating system images built for bare metal deployments.
+    :param image_url: URL where the image resides
+    :type image_url: str
+    """
+    image_url = image_url or os.environ.get(
+        'TEST_MAGNUM_QCOW2_IMAGE', None)
+    image_name = 'fedora-coreos'
```

Comment:
Make the image name a global variable in this setup file, so you can import it in the Magnum tests, and use it.